cartography/intel/aws/rds.py

```diff
@@ -98,6 +105,132 @@ def load_rds_instances(neo4j_session, data, region, current_aws_account_id, aws_
+    instance in a VPC, you must select a DB subnet group. Amazon RDS uses that DB subnet group and your preferred
+    Availability Zone to select a subnet and an IP address within that subnet to associate with your DB instance.`
+    """
+    attach_subnets_to_sng = """
+    MATCH(subnet:EC2Subnet{subnetid:{SubnetIdentifier}}),
+    (sng:DBSubnetGroup{id:{DBSubnetGroupName}})
+    MERGE(sng)-[r:RESOURCE]->(subnet)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag},
+    subnet.availability_zone = {SubnetAvailabilityZone}
+    """
+    # TODO: Note: RDS data has subnet availability zone data, but I don't think that this is the right place to
+    # TODO: update it.  We should update the availability zone data in the EC2 ingestion part.
+    if not db_subnet_group.get('DBSubnetGroupName'):
+        logger.debug("Expected DBSubnetGroup to have a DBSubnetGroupName but it doesn't.  Here is the object: %r",
+                     db_subnet_group)
+    else:
+        for sn in db_subnet_group.get('Subnets', []):
+            subnet_id = sn.get('SubnetIdentifier', None)
+            if not subnet_id:
+                logger.debug("Expected Subnet to have a SubnetIdentifier but it doesn't. Here is the object: %r",
+                             db_subnet_group)
+            else:
+                neo4j_session.run(
+                    attach_subnets_to_sng,
+                    SubnetIdentifier=subnet_id,
+                    DBSubnetGroupName=db_subnet_group.get('DBSubnetGroupName'),
+                    aws_update_tag=aws_update_tag,
+                    SubnetAvailabilityZone=sn.get('SubnetAvailabilityZone', {}).get('Name', None)
+                )
+
+
+def _attach_ec2_security_groups(neo4j_session, instance, aws_update_tag):
+    """
+    Attach an RDS instance to its EC2SecurityGroups
+    """
+    attach_rds_to_group = """
+    MATCH (rds:RDSInstance{id:{RdsArn}}),
+    (sg:EC2SecurityGroup{groupid:{GroupId}})
+    MERGE (rds)-[m:MEMBER_OF_EC2_SECURITY_GROUP]->(sg)
+    ON CREATE SET m.firstseen = timestamp()
+    SET m.lastupdated = {aws_update_tag}
+    """
+    if not instance.get('DBInstanceArn'):
+        logger.debug("Expected RDSInstance to have a DBInstanceArn but it doesn't.  Here is the object: %r", instance)
+    else:
+        for group in instance.get('VpcSecurityGroups', []):
+            neo4j_session.run(
+                attach_rds_to_group,
+                RdsArn=instance.get('DBInstanceArn'),
+                GroupId=group.get('VpcSecurityGroupId'),
+                aws_update_tag=aws_update_tag
+            )
+
+
+def _attach_read_replicas(neo4j_session, read_replicas, aws_update_tag):
+    """
+    Attach read replicas to their source instances
+    """
+    attach_replica_to_source = """
+    MATCH (replica:RDSInstance{id:{ReplicaArn}}),
+    (source:RDSInstance{db_instance_identifier:{SourceInstanceIdentifier}})
+    MERGE (replica)-[r:IS_READ_REPLICA_OF]->(source)
```

Comment:
BLOCK - missing cleanup

Reply:
RDS Instances are all cleaned up [here](https://github.com/lyft/cartography/pull/10/files#diff-db18c81378ce7f90e09bd4f343906bedR9).

Reply:
refering to IS_READ_REPLICA_OF